networks/rpc/types.go

```diff
@@ -170,3 +179,117 @@ func (bn *BlockNumber) UnmarshalJSON(data []byte) error {
 func (bn BlockNumber) Int64() int64 {
 	return (int64)(bn)
 }
+
+func (bn BlockNumber) Uint64() uint64 {
+	return (uint64)(bn)
+}
+
+type BlockNumberOrHash struct {
+	BlockNumber      *BlockNumber `json:"blockNumber,omitempty"`
+	BlockHash        *common.Hash `json:"blockHash,omitempty"`
+	RequireCanonical bool         `json:"requireCanonical,omitempty"`
+}
+
+func (bnh *BlockNumberOrHash) UnmarshalJSON(data []byte) error {
+	type erased BlockNumberOrHash
+	e := erased{}
+	err := json.Unmarshal(data, &e)
+	if err == nil {
+		if e.BlockNumber != nil && e.BlockHash != nil {
+			return fmt.Errorf("cannot specify both BlockHash and BlockNumber, choose one or the other")
+		}
+		bnh.BlockNumber = e.BlockNumber
+		bnh.BlockHash = e.BlockHash
+		bnh.RequireCanonical = e.RequireCanonical
+		return nil
+	}
+
+	var decimalInput uint64
+	err = json.Unmarshal(data, &decimalInput)
+	if err == nil {
+		bn := BlockNumber(decimalInput)
+		bnh.BlockNumber = &bn
+		return nil
+	}
+
+	var input string
+	err = json.Unmarshal(data, &input)
+	if err != nil {
+		return err
+	}
+	switch input {
+	case "earliest":
+		bn := EarliestBlockNumber
+		bnh.BlockNumber = &bn
+		return nil
+	case "latest":
+		bn := LatestBlockNumber
+		bnh.BlockNumber = &bn
+		return nil
+	case "pending":
+		bn := PendingBlockNumber
+		bnh.BlockNumber = &bn
+		return nil
+	default:
+		if len(input) == 66 {
+			hash := common.Hash{}
+			err := hash.UnmarshalText([]byte(input))
+			if err != nil {
+				return err
+			}
+			bnh.BlockHash = &hash
+			return nil
+		} else {
+			blckNum, err := hexutil.DecodeUint64(input)
+			if err != nil {
+				return err
+			}
+			if blckNum > math.MaxInt64 {
+				return fmt.Errorf("blocknumber too high")
+			}
+			bn := BlockNumber(blckNum)
+			bnh.BlockNumber = &bn
+			return nil
+		}
+	}
+}
+
+func (bnh *BlockNumberOrHash) Number() (BlockNumber, bool) {
+	if bnh.BlockNumber != nil {
+		return *bnh.BlockNumber, true
+	}
+	return BlockNumber(0), false
+}
+
+func (bnh *BlockNumberOrHash) Hash() (common.Hash, bool) {
+	if bnh.BlockHash != nil {
+		return *bnh.BlockHash, true
+	}
+	return common.Hash{}, false
+}
+
+func (bnh *BlockNumberOrHash) NumberOrHashString() (string, bool) {
+	if bnh.BlockHash != nil {
+		return bnh.BlockHash.String(), true
+	}
+	if bnh.BlockNumber != nil {
+		return fmt.Sprintf("#%v", bnh.BlockNumber.Int64()), true
+	}
+	return "", false
+}
+
+func NewBlockNumberOrHashWithNumber(blockNr BlockNumber) BlockNumberOrHash {
```

Comment:
How about using the same naming rule with Ethereums?

Reply:
There is no "New" in that method naming of ethereum.
But klaytn other code namings often have "New", so I added it for consistency...
I don't know which is more important, consistency or compatibility.
Frankly, I don't know the team's convention yet. :cry: :cry:

Reply:
I think it was my suggestion.. sorry for the confusion! If ethereum compatibility is more important, it would be better to go back.